submission/persistence-filesystem/src/main/kotlin/ac/uk/ebi/biostd/persistence/filesystem/fire/FireFilesService.kt

```diff
@@ -34,32 +34,33 @@ class FireFilesService(
     ): FireFile {
         return when (val path = fireFile.firePath) {
             expectedPath -> fireFile
-            null -> setMetadata(fireFile.fireId, fireFile, expectedPath, fireFile.published)
+            null -> setMetadata(fireFile.fireId, fireFile, expectedPath)
             else -> {
                 val file = requireNotNull(client.downloadByPath(path)) { "Could not download file with path $path" }
                 val saved = client.save(file, fireFile.md5, fireFile.size)
-                setMetadata(saved.fireOid, fireFile, expectedPath, false)
+                setMetadata(saved.fireOid, fireFile, expectedPath)
             }
         }
     }
 
     private suspend fun getOrCreate(file: NfsFile, expectedPath: String): FireFile {
-        val matches = client.findByMd5(file.md5)
-        val apiFile = matches.find { it.path == expectedPath }
-            ?: matches.find { it.path == null }
-            ?: client.save(file.file, file.md5, file.size)
+        val previousFile = client.findByPath(expectedPath)
+        val apiFile = when {
+            previousFile != null && previousFile.objectMd5 == file.md5 -> previousFile
+            else -> client.save(file.file, file.md5, file.size)
```

Comment:
this does not look correct for me if the file md5 is not the same we wll not succed to save file in the same path so there is not point to saved it. So IMO if file is there at this point is because is because is the same otherwize it should have clean in a previous stage.